### DISCUSSION TITLE
How to draw an arc?

### DISCUSSION THREAD

**Discussion by minystreem56** [MonoGame/MonoGame#8291] [Graphics]
I need to draw an arc for my game, how would I draw a cubic spline?

**Comment by Apostolique** [comment]
One way is to use a SDF to draw an arc. Inigo Quilez has info on his site: <https://iquilezles.org/articles/distfunctions2d/>.

You can find an example of how to draw shapes in my library: <https://github.com/Apostolique/Apos.Shapes>. I don't support arcs yet though.

**Comment by EuReinoso** [comment]
You can draw an Array/List of Lines.

This guy has a playlist teaching concepts for drawing shapes in monogame: https://www.youtube.com/watch?v=BQk0VuhgONE&list=PLSlpr6o9vURx4vjomFuwrFhvhV1nhJ_Jc&index=29

I recommend the episodes: 06 - Shape Drawing, 07 Drawing Lines and 12 - Drawing Polygons

After that you need to apply the calculations to draw the arc you want, the calculation will depend on the type of arc.
